### PR TITLE
Add helpers to disable progress bars globally + tests

### PR DESCRIPTION
**Pull request by Wauplin**
Solve https://github.com/huggingface/huggingface_hub/issues/978 (see API discussion in issue).

@sgugger Since both `HF_HUB_DISABLE_PROGRESS_BARS` env variable and `disable_progress_bars()`/`enable_progress_bars()` helpers are defined, there might be a conflict at some point if not correctly handled. For example a library, could call `disable_progress_bars()` even though the user has set `HF_HUB_DISABLE_PROGRESS_BARS=0`. Current solution is that `disable_progress_bars()` has higher priority but I'm not 100% sure it's clever.

The other solution would be the opposite: if `HF_HUB_DISABLE_PROGRESS_BARS` is set (either 0 or 1), then `disable_progress_bars`/`enable_progress_bars` are ignored in the code with a warning (`"cannot programmatical disable progress bars as verbosity has been set by environment variable"`). 

What is your opinion about it ? I feel that both are correct, it's more a usage preference.

**EDIT:** after discussion below, we went for solution 2 (environment variable has priority).

### CODE REVIEW

File: src/huggingface_hub/constants.py [hunk 1]
```diff
@@ -68,8 +74,6 @@
 
 HUGGINGFACE_HUB_CACHE = os.getenv("HUGGINGFACE_HUB_CACHE", default_cache_path)
 
-HF_HUB_OFFLINE = os.environ.get("HF_HUB_OFFLINE", "AUTO").upper()
-if HF_HUB_OFFLINE in ENV_VARS_TRUE_VALUES:
-    HF_HUB_OFFLINE = True
-else:
-    HF_HUB_OFFLINE = False
```
Comment (Wauplin) [inline src/huggingface_hub/constants.py:75]: I suspect that `"AUTO"` never really worked as expected here. I refactored it to make it more readable keeping exactly the same functionality but if we want `"AUTO"` to be considered as True, we need to use  `ENV_VARS_TRUE_AND_AUTO_VALUES` (so `_is_true_or_auto()`).

File: src/huggingface_hub/utils/tqdm.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    # Re-enable progress bars globally
+    enable_progress_bars()
+
+    # Progress bar will be shown !
+    for _ in tqdm(range(5)):
+       do_something()
+    ```
+"""
+from tqdm.auto import tqdm as _tqdm
```
Comment (sgugger) [inline src/huggingface_hub/utils/tqdm.py:55]: Complete nit but when monkey-patching things, I like to prefix the old object with `old_`
```suggestion
from tqdm.auto import tqdm as old_tqdm
```

**Review by sgugger — APPROVED** [review]
Thanks a lot for working on this!

I think the env variable should get priority over programmatic functions. My reasoning being that if the user goes out of their way to set it on the env level, it's really clear they want that thing all the time. But it's just my 2 cents and going the other way around also works.

**Comment by Wauplin** [comment]
> I think the env variable should get priority over programmatic functions. My reasoning being that if the user goes out of their way to set it on the env level, it's really clear they want that thing all the time. But it's just my 2 cents and going the other way around also works.

Makes sense. That's what I also started to think when writing the PR text actually but didn't want to change the code before having a second opinion :D 
Will make the change.

**Comment by Wauplin** [comment]
@sgugger I made the changes to give priority to env variable in commit https://github.com/huggingface/huggingface_hub/pull/987/commits/b47378a7f6b9f10906bb5d817c78255e8ca37598. 
Could you quickly review it so that I can merge ? Thanks in advance 🙏

**Pull request merged**